### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by hammaazarok**
## in this milestone we achieved the following :
- Add Unit tests for the following Classes
   - Book
   - Person
   - Student
   - Teacher
   - Classroom
   - Rental
   - Decorators for Person Class

### CODE REVIEW

File: unit_tests_spec.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  let(:trimmed) do
+    TrimmerDecorator.new(person).correct_name
+  end
+
+  describe '#Capitalize' do
+    it 'Capitalize persons name' do
+      expect(capitalized).to eq('John doe')
+    end
+  end
+
```
Comment (tufoinnkuo10) [inline unit_tests_spec.rb]: - [x] please kindly add test to check if persons name and age is present

File: unit_tests_spec.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
+require_relative 'book'
```
Comment (tufoinnkuo10) [inline unit_tests_spec.rb]: - [x] great job with the test. please ensure that All test files should be placed in a different directory (e.g /spec) . that is There should be one file per class.
eg for testing person class . it should be inside 
`spec/person_spec.rb`

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
## Hi @TEAM 

There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally


## Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: tests/rental.spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require_relative '../rental'
+
```
Comment (tresorsawasawa) [inline tests/rental.spec.rb]: - [x] As mentioned by the previous reviewer and ad required for the project, kindly make sure  all the test files are placed in a directory named `spec` and not `tests` i.e. you can only edit the `tests` to `spec`

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @hammaazarok 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**